File: client/src/main/java/io/theam/client/commands/CustomerListCommand.java
package io.theam.client.commands;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.github.rvesse.airline.annotations.Command;
import io.theam.client.service.RestClient;
import io.theam.model.Customer;

import java.util.Collection;

@Command(name = "list", description = "Get the list of all customers")
public class CustomerListCommand extends BaseCommand {

    @Override
    public void doRun() {

        Collection<Customer> customers = null;
        customers = new RestClient(username, password).getCustomers();

        try {
            System.out.println(pretty_print_json.writeValueAsString(customers));
        } catch (JsonProcessingException e) {
            new RuntimeException(e);
        }
        System.out.println("---------");
        System.out.println("Number of customers: " + Integer.toString(customers.size()));
    }
}

File: client/src/main/kotlin/io/theam/client/commands/customers/CustomerListCommand.kt
package io.theam.client.commands.customers

import com.github.rvesse.airline.annotations.Command
import io.theam.client.commands.BaseCommand
import io.theam.client.service.bodyOf
import io.theam.client.service.getEntity
import io.theam.client.service.withUrl
import io.theam.model.api.CustomerResponse
import org.springframework.core.ParameterizedTypeReference

@Command(name = "list", description = "Get the list of all customers")
class CustomerListCommand : BaseCommand() {

    private val customers: Collection<CustomerResponse>
        get() = bodyOf( restClient withUrl
                "$host/customers" getEntity
                    object : ParameterizedTypeReference<Collection<CustomerResponse>>() {})

    public override fun doRun() =
            customers.let { listOf(
                    pretty_print_json.writeValueAsString(it),
                    "------------",
                    "Number of customers: ${it.size}")
                    .forEach {println(it)} }
}
